checker.cpp

```diff
@@ -1,17 +1,97 @@
+// ConsoleApplication3.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+
+#include <iostream>
+
 #include <assert.h>
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
+using namespace std;
+
+float MinBPM[] = { 70 };
+float MaxBPM[] = { 150 };
+float MinSpo2[] = { 90 };
+float MinRespRate[] = { 30 };
+float MaxRespRate[] = { 95 };
+
+class Alert
+{
+public:
+	virtual void SendAlert(const char* vitalName, const char* Message) = 0;
+};
+class SmsAlert : public Alert
+{
+public:
+	void SendAlert(const char* vitalName, const char* Message)
+	{
+		std::cout << vitalName << ": " << Message << endl;
+	}
+};
+class SoundAlert : public Alert
+{
+public:
+	void SendAlert(const char* vitalName, const char* Message)
+	{
+		std::cout << vitalName << ": " << Message << endl;
+	}
+};
+
+
+void checkRespRate(Alert *alert, float value)
+{
```

Comment:
The check... functions repeat in a pattern. Consequence is multiple places to fix... e.g., if there should not be any 'normal' alert (because no attention is needed), you will need to modify the code in three places. Reduce the duplication.